docs/website/docs/general-usage/data-enrichments/currency_conversion_data_enrichment.md

```diff
@@ -0,0 +1,267 @@
+   			  that can be changed
+           The API key is retrieved from the `dlt` secrets.
+       """
+
+       # Hardcoded base and target currencies
+       base_currency = "USD"
+       target_currency = "EUR"
+
+       # Retrieve the API key from DLT secrets
+       api_key = dlt.secrets.get("sources.api_key")
+
+       # Initialize or retrieve the state for currency rates
+       rates_state = dlt.current.resource_state().setdefault("rates", {})
+       currency_pair_key = f"{base_currency}-{target_currency}"
+       currency_pair_state = rates_state.setdefault(currency_pair_key, {
+           "last_update": datetime.min,
+           "rate": None
+       })
+
+       # Update the exchange rate if it's older than 12 hours
+       if (
+       currency_pair_state.get("rate") is None
+       or (datetime.utcnow() - currency_pair_state["last_update"] >= timedelta(hours=12))):
+           url = f"https://v6.exchangerate-api.com/v6/{api_key}/pair/{base_currency}/{target_currency}"
+           response = requests.get(url)
+           if response.status_code == 200:
+               data = response.json()
+               currency_pair_state.update({
+                   "rate": data.get("conversion_rate"),
+                   "last_update": datetime.fromtimestamp(data.get("time_last_update_unix"))
+               })
+               print(f"The latest rate of {data.get('conversion_rate')} for the currency pair {currency_pair_key} is fetched and updated.")
+           else:
+               raise Exception(f"Error fetching the exchange rate: HTTP {response.status_code}")
+
+       # Convert the amount using the retrieved or stored exchange rate
+       amount = record['device_price_usd']
+       rate = currency_pair_state["rate"]
+       yield {
+           "actual_amount": amount,
+           "base_currency": base_currency,
+           "converted_amount": round(amount * rate, 2),
+           "target_currency": target_currency,
+           "rate": rate,
+             "rate_last_updated": currency_pair_state["last_update"],
+       }
+   ```
+
+1. Next, follow the instructions in
+   [Destinations](../../dlt-ecosystem/destinations/duckdb.md) to add credentials for
+   your chosen destination. This will ensure that your data is properly routed to its final
+   destination.
+
+### 3. Create your pipeline
+
+1. In creating the pipeline, the `converted_amount` can be used in the following ways:
+
+   - Add map function
+   - Transformer function
+
+   The `dlt` library's `transformer` and `add_map` functions serve distinct purposes in data
+   processing.
+
+   `Transformers` used to process a resource and are ideal for post-load data transformations in a
+   pipeline, compatible with tools like `dbt`, the `dlt SQL client`, or Pandas for intricate data
+   manipulation. To read more:
+   [Click here.](../../general-usage/resource.md#process-resources-with-dlttransformer)
```

Comment:
not really correct, e.g. how we can use dbt and transformers? and it's not a post-load transformations, you just can  iterate over another resource and manipulate the data.

Reply:
yes, corrected